### ISSUE TITLE
Quasi Arbitrary Resolution

### ISSUE THREAD

**Issue by pmjdebruijn** [fabiangreffrath/crispy-doom#73]
As crispy is currently limited to multiples of 640x400/640x480, there are some awkward resolutions that don't fit well.

For example 1366x768 is extremely common on laptops, 1280x800 won't fit, and 640x400 is a tiny window.

My suggestion would be to for example take the display height, and start dividing it, so we get a value between 200-400, in this particular case: 768/2=384
Then calculate the respective height (aspect corrected): 384/240*320=512

Then render 512x384, and output 1024x768 (pixel doubled)

I'm not sure how well this fits into the doom renderer though.

**Comment by fabiangreffrath** [comment]
This is outright impossible with Doom's own rendering algorithm.

However, there is work ongoing (with good progress as of recently) in Chocolate Doom's sdl2-branch that will scale the original 320x200 resolution into whichever resolution the target window or screen has. Once this effort is finished there, it will get merged into Crispy Doom (and extended to 640x480 of course). This branch is also the prerequisite for the high-color rendering branch that I have in the waiting, so I am really behind this. ;)

**Comment by fabiangreffrath** [comment]
I think this is fixed with the switch to SDL2 in Crispy Doom 5.0 and onward.
